### PR TITLE
[web3torrent] Clean up File component

### PR DESCRIPTION
**Pull request by kerzhner**
- Use `function` syntax when possible. I am finding a constant that defines a function (with Typescript types sprinkled in there) pretty hard to parse. If others agree, please use `function` syntax going forward (when possible).
- `File` component has a `useContext` hook now, so we can remove `Web3TorrentContext.Consumer`. In general, I might be a good idea to minimize the logic inside jsx and instead use jsx primarily to describe UI.

### CODE REVIEW

File: packages/web3torrent/src/pages/file/File.tsx [hunk 1]
```diff
@@ -46,41 +46,33 @@ const File: React.FC<Props> = props => {
...
-                  mySigningAddress={me}
-                />
-              ) : (
-                false
-              )}
-            </>
-          );
-        }}
-      </Web3TorrentContext.Consumer>
+      <>
```
Comment (spersico) [inline packages/web3torrent/src/pages/file/File.tsx:62]: this react fragment is not necessary

File: packages/web3torrent/src/pages/file/File.tsx [hunk 2]
```diff
@@ -46,41 +46,33 @@ const File: React.FC<Props> = props => {
...
+        <br />
+        {showBudget ? (
+          <SiteBudgetTable
+            budgetCache={budgetCache}
+            channelCache={channelCache}
+            mySigningAddress={me}
+          />
+        ) : (
+          false
+        )}
```
Comment (spersico) [inline packages/web3torrent/src/pages/file/File.tsx:73]: this could be 
```showBudget && (<SiteBudgetTable ....``` to avoid the `: (false)`. 
`false` values don't get shown in the render.

File: packages/web3torrent/src/pages/file/File.tsx [hunk 3]
```diff
@@ -46,41 +46,33 @@ const File: React.FC<Props> = props => {
...
+      </>
+      }
```
Comment (spersico) [inline packages/web3torrent/src/pages/file/File.tsx:75]: Ups! You missed one :stuck_out_tongue: 
image

File: packages/web3torrent/src/pages/file/File.tsx [hunk 4]
```diff
@@ -46,41 +46,33 @@ const File: React.FC<Props> = props => {
...
       {torrent.status === Status.Idle ? (
```
Comment (spersico) [inline packages/web3torrent/src/pages/file/File.tsx:76]: this could also be `torrent.status && Status.Idle ? (....`

File: packages/web3torrent/src/pages/file/File.tsx (hunk 1 above)
Comment (kerzhner) [reply packages/web3torrent/src/pages/file/File.tsx]: 37e02c957

File: packages/web3torrent/src/pages/file/File.tsx (hunk 2 above)
Comment (kerzhner) [reply packages/web3torrent/src/pages/file/File.tsx]: 37e02c957

File: packages/web3torrent/src/pages/file/File.tsx (hunk 3 above)
Comment (kerzhner) [reply packages/web3torrent/src/pages/file/File.tsx]: Good catch! 37e02c957

File: packages/web3torrent/src/pages/file/File.tsx (hunk 4 above)
Comment (kerzhner) [reply packages/web3torrent/src/pages/file/File.tsx]: 37e02c957
